### PR TITLE
Update Ansible Collections

### PR DESCRIPTION
**Pull request by theishshah**
**Description of the change:**

Apply patch from #4190 and update the comparison operator used in the templates to match downstream.

### CODE REVIEW

**Review by estroz — CHANGES_REQUESTED** [review]
On second thought, for reproducibility purposes it's probably best to pin major and minor versions, `==1.1.*`.

Also the [molecule tests](https://github.com/operator-framework/operator-sdk/blob/v1.4.0/hack/tests/e2e-ansible-molecule.sh#L23) need to be updated.

File: hack/tests/e2e-ansible-molecule.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ pip3 install pyasn1==0.4.7 pyasn1-modules==0.2.6 idna==2.8 ipaddress==1.0.23
 pip3 install cryptography==3.3.2 molecule==3.0.2
 pip3 install ansible-lint yamllint
 pip3 install docker==4.2.2 openshift==0.11.2 jmespath
-ansible-galaxy collection install 'community.kubernetes:<1.0.0'
+ansible-galaxy collection install 'community.kubernetes:>=1.1.0'
```
Comment (estroz) [inline hack/tests/e2e-ansible-molecule.sh:23]:
```suggestion
ansible-galaxy collection install 'community.kubernetes:==1.1.1'
```

File: internal/plugins/ansible/v1/scaffolds/internal/templates/requirements.go [hunk 2]
```diff
@@ -34,6 +34,6 @@ func (f *RequirementsYml) SetTemplateDefaults() error {
 const requirementsYmlTmpl = `---
 collections:
   - name: community.kubernetes
-    version: "<1.0.0"
+    version: "==1.1.*"
```
Comment (estroz) [inline internal/plugins/ansible/v1/scaffolds/internal/templates/requirements.go:37]:
```suggestion
    version: "==1.1.1"
```

File: testdata/ansible/memcached-operator/requirements.yml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ---
 collections:
   - name: community.kubernetes
-    version: "<1.0.0"
+    version: "==1.1.*"
```
Comment (estroz) [inline testdata/ansible/memcached-operator/requirements.yml:4]:
```suggestion
    version: "==1.1.1"
```

**Review by estroz — CHANGES_REQUESTED** [review]
Looks like ansible collections don't like pip semver syntax. Better to pin to a specific minor version for full reproducibility anyway.

Still needs a changelog.

**Review by estroz — APPROVED** [review]
/lgtm

**Review by jmrodri — APPROVED** [review]
/lgtm

File: internal/plugins/ansible/v1/scaffolds/internal/templates/requirements.go [hunk 4]
```diff
@@ -34,7 +34,6 @@ func (f *RequirementsYml) SetTemplateDefaults() error {
 const requirementsYmlTmpl = `---
 collections:
   - name: community.kubernetes
-    version: "1.1.1"
-  - name: operator_sdk.util
-    version: "0.1.0"
```
Comment (estroz) [inline internal/plugins/ansible/v1/scaffolds/internal/templates/requirements.go:39]: Why did you remove this `version`?

Comment (Adirio) [reply internal/plugins/ansible/v1/scaffolds/internal/templates/requirements.go:39]: I just found this too, probably it was just a non-properly resolved merge conflict as that changed was recently merged in #4529.
